### PR TITLE
feat: merge PD incidents automatically

### PR DESCRIPTION
**Pull request by whereswaldon**
Signed-off-by: Chris Waldon <cwaldon@redhat.com>

### CODE REVIEW

File: pkg/common/pagerduty/pagerduty.go [hunk 1]
```diff
@@ -25,3 +27,110 @@ func (pdc Config) FireAlert(details pd.V2Payload) (*pd.V2EventResponse, error) {
...
+				return fmt.Errorf("handler failed: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+// Alerts uses the provided client to retrieve all Alerts associated with the provided
+// incident, calling the provided handler function on each alert.
+func Alerts(c *pd.Client, incident pd.Incident, options pd.ListIncidentAlertsOptions, handler func(pd.IncidentAlert) error) error {
```
Comment (jeefy) [inline pkg/common/pagerduty/pagerduty.go:114]: Is `Alerts` used anywhere? I can't find anything in this PR referencing it, maybe I'm blind.

**Review by jeefy — CHANGES_REQUESTED** [review]
One question otherwise super lgtm :rainbow:

File: pkg/common/pagerduty/pagerduty.go (hunk 1 above)
Comment (whereswaldon) [reply pkg/common/pagerduty/pagerduty.go:114]: I originally used it, then concluded that I didn't need it to perform this particular task. However, it is a working, tested implementation of something that I expect to need if we make this policy more sophisticated, so I thought I'd leave it in there. Happy to remove it for now if you'd prefer.

**Pull request merged**